src/porepy/models/run_models.py

```diff
@@ -101,7 +101,7 @@ def time_step() -> None:
             + f" of {model.time_manager.time_final:.1e}"
             + f" with time step {model.time_manager.dt:.1e}"
         )
-        solver.solve(model)
+        return solver.solve(model)
```

Comment:
I am in general a fan of collecting the performance status. This can be done in different ways, with different purposes. I see, how you want to use here. To me this looks OK. I would suggest adding a short comment mentioning the content, i.e., convergence-status, so one does not have to scroll to look for the purpose.

Reply:
Done.